src/mlpack/methods/edge_boxes/feature_extraction.hpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file feature_extraction.hpp
+ * @author Nilay Jain 
+ *
+ * Feature Extraction for the edge_boxes algorithm.
+ */
+#ifndef MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_HPP
+#define MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_HPP
+//#define INF 999999.9999
+//#define EPS 1E-20
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace structured_tree {
+
+template <typename MatType = arma::mat, typename CubeType = arma::cube>
+class StructuredForests
+{
+
+ public:
+
+  static constexpr double eps = 1e-20;
+
+  std::map<std::string, size_t> options;
+  
+  StructuredForests(const std::map<std::string, size_t> inMap);
+  
```

Comment:
Mmm do you think, it would be reasonable to use a parameter for every setting, instead of a map? Using a map is pretty uncommon in the mlpack codebase.